src/stream/ngx_stream_mruby_connection.c

```diff
@@ -39,10 +40,8 @@ static mrb_value ngx_stream_mrb_connection_init(mrb_state *mrb, mrb_value self)
   ngx_stream_mruby_upstream_context *ctx;
   ngx_stream_upstream_main_conf_t *umcf;
   ngx_stream_upstream_srv_conf_t **usp;
-  ngx_stream_mruby_internal_ctx_t *ictx = mrb->ud;
-  ngx_stream_session_t *s = ictx->s;
 
-  mrb_get_args(mrb, "o", &upstream);
+  mrb_get_args(mrb, "|o", &upstream);
```

Comment:
Why did you use optimal arguments?

Reply:
I think it is not necessary to specify an upstream server.
What do you think?

Reply:
You mean that even if we don't specify an upstream configuration, we want to use instance methods like `local_ip`, right? OK, fair enough.

Reply:
exactly right!